contracts/BaseWrapper.sol

```diff
@@ -153,11 +153,10 @@ abstract contract BaseWrapper {
         VaultAPI _bestVault = bestVault();
 
         if (pullFunds) {
-            if (amount != DEPOSIT_EVERYTHING) {
-                SafeERC20.safeTransferFrom(token, depositor, address(this), amount);
-            } else {
-                SafeERC20.safeTransferFrom(token, depositor, address(this), token.balanceOf(depositor));
+            if (amount == DEPOSIT_EVERYTHING) {
+                amount = token.balanceOf(depositor);
```

Comment:
If I understand this change correctly, the update of `amount` here will ensure the `safeApprove` call is skipped on line 163 based on the current allowance

Reply:
Yes that's the goal